src/fonts.jl

```diff
@@ -0,0 +1,226 @@
+           end
+           return unsafe_string(output_str[])
+end
+
+
+"""
+     extract_fc_attr_int(pat::Fontconfig.Pattern, attr::String)::Int
+
+Extracts the FontConfig attribute, whose name is given by `attr`, from a `Fontconfig.Pattern`.
+Returns a String.
+"""
+function extract_fc_attr_int(pat::Fontconfig.Pattern, attr::String; num::Int = 0)
+       local output_int = Ref{Cint}()
+       success = @ccall Fontconfig.Fontconfig_jll.libfontconfig.FcPatternGetInteger(pat.ptr::Ptr{Cvoid}, attr::Cstring, num::Cint, output_int::Ptr{Cint})::Cint
+       if success != 0
+           error("The operation did not succeed!  Fontconfig returned the error code $success when extracting the $(num)th entry of attribute $attr from pattern $pat.")
+       end
+       return output_int[]
+end
+
+# Now for the real meat - interfacing FreeTypeAbstraction and Fontconfig
+
+"""
+    ftfont_from_fc_pattern(pattern::Fontconfig.Pattern)::FreeTypeAbstraction.FTFont
+
+Construct an FTFont from the provided `pattern`, assuming the `file` and `index`
+properties are present in `pattern`.
+"""
+function ftfont_from_fc_pattern(pattern::Fontconfig.Pattern)::FreeTypeAbstraction.FTFont
+    # Extract the file path and the index of the font within that file
+    # from the Pattern.
+    # WARNING: This assumes that they are both present!
+    file = extract_fc_attr_string(pattern, "file")
+    index = extract_fc_attr_int(pattern, "index")
+    # Construct a FTFont, given the above information.
+    return FreeTypeAbstraction.FTFont(FreeTypeAbstraction.newface(file, index))
+end
+
+# This performs a search using Fontconfig and loads the result
+# as a Makie-compatible NamedTuple.
+
+"""
+    populate_font_family(family::String; additional_params_for_fontconfig...)
+
+Returns a NamedTuple of styles and loaded FreeType fonts available for family names.
+This returns a NamedTuple with symbolic keys and `FreeTypeAbstraction.FTFont` values.
+
+!!! danger
+    The keys will be different for each font family!  This will require normalization
+    of some kind.
+"""
+function populate_font_family(family::String; additional_params_for_fontconfig...)
+    # Create a pattern to search for
+    pattern = Fontconfig.Pattern(; family = family, additional_params_for_fontconfig...)
+    # Use Fontconfig to generate a list of styles from that pattern.
+    font_list = fclist(pattern, ["family", "style", "file", "index"])
+    # Extract the style names (regular, light, bold, oblique, etc)
+    styles = extract_fc_attr_string.(font_list, "style")
+    # Return a NamedTuple
+    return (; (Symbol.(lowercase.(styles)) .=> ftfont_from_fc_pattern.(font_list))...)
+end
+
+
+# Some convenient ways to get styles
+
+function is_bold(font::FreeTypeAbstraction.FTFont)
+    return font.style_flags & 1
+end
+
+function is_italic_or_oblique(font::FreeTypeAbstraction.FTFont)
+    return font.style_flags & 2
+end
+
+
+function search_font(family::String, style::String = "regular"; exact = false)::NativeFont
```

Comment:
I could move a slightly refactored form of this to Fontconfig, where the user can pass various substitutions or required elements that they want.